library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -953,6 +962,8 @@ private Calendar getTimeFromPoint(float x, float y){
                         - mHeaderRowPadding * 2 - mTimeTextHeight/2 - mHeaderMarginBottom;
                 int hour = (int)(pixelsFromZero / mHourHeight);
                 int minute = (int) (60 * (pixelsFromZero - hour * mHourHeight) / mHourHeight);
+                minute -= Math.min(minute,  (mNewEventLengthInMinutes / 2));
```

Comment:
I think this shouldn't always happen, this method is also used at other places where the length of new events doesn't make any difference.